### PR TITLE
COMP: Fix OpenJPEG build error with Visual Studio 16.9

### LINKED ISSUE #1967: ITK failed to build due to error C2169: 'lrintf': intrinsic function, cannot be defined with MSVC on windows
Hi All,

ITK failed to build due to error C2169: 'lrintf': intrinsic function, cannot be defined with MSVC on windows. It can be reproduced on master branch latest commit 26079ef.  Could you please help look at this? Thanks a lot!

Repro Steps:
1. git clone https://github.com/InsightSoftwareConsortium/ITK.git F:\gitP\InsightSoftwareConsortium\ITK
2. Open VS2019 x64 command prompt as admin and browse to F:\gitP\InsightSoftwareConsortium\ITK
3. Apply itk_bug623130.patch
4. mkdir build_amd64 and cd build_amd64
5. cmake -G "Visual Studio 16 2019" -A x64 -DCMAKE_SYSTEM_VERSION=10.0.18362.0 -DBUILD_SHARED_LIBS=OFF -DBUILD_TESTING=ON -DBUILD_EXAMPLES=OFF ..
6. set _CL_=/D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING %_CL_%
7. msbuild /m /p:Platform=x64 /p:Configuration=Release All_BUILD.vcxproj /t:Rebuild

Build_log:
[build.log](https://github.com/InsightSoftwareConsortium/ITK/files/5141200/build.log)
itk_bug623130.patch:
[itk_bug623130.zip](https://github.com/InsightSoftwareConsortium/ITK/files/5141203/itk_bug623130.zip)

ErrorMessage:
188>F:\gitP\InsightSoftwareConsortium\ITK\Modules\ThirdParty\OpenJPEG\src\openjpeg\opj_includes.h(106,35): error C2169: 'lrintf': intrinsic function, cannot be defined [F:\gitP\InsightSoftwareConsortium\ITK\build_amd64\Modules\ThirdParty\OpenJPEG\src\openjpeg\itkopenjpeg.vcxproj]

### PR DESCRIPTION
**Pull request by jamesobutler**
This closes #1967.

Co-authored-by: @spacelg <42860526+spacelg@users.noreply.github.com>

Fixes error:

> Error	C2169	'lrintf': intrinsic function, cannot be defined [M:\Dev\Plus\SPc\itk-bin\Modules\ThirdParty\OpenJPEG\src\openjpeg\itkopenjpeg.vcxproj]	itk	M:\Dev\Plus\SPc\itk\Modules\ThirdParty\OpenJPEG\src\openjpeg\opj_includes.h	106

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

File: Modules/ThirdParty/OpenJPEG/src/openjpeg/opj_includes.h [hunk 1]
```diff
@@ -92,7 +92,7 @@ Most compilers implement their own version of this keyword ...
 
 /* MSVC 64bits doesn't support _asm */
 #if !defined(_WIN64)
-static INLINE long lrintf(float f){
+static INLINE long opj_lrintf(float x){
```
Comment (dzenanz) [inline Modules/ThirdParty/OpenJPEG/src/openjpeg/opj_includes.h:95]: I don't think we should change the name of the parameter, as it is referenced by the assembly code. It should stay `f`.

Comment (jamesobutler) [reply Modules/ThirdParty/OpenJPEG/src/openjpeg/opj_includes.h:95]: Good catch. I'll issue a change

Comment (jamesobutler) [reply Modules/ThirdParty/OpenJPEG/src/openjpeg/opj_includes.h]: Update has been pushed

Comment (dzenanz) [reply Modules/ThirdParty/OpenJPEG/src/openjpeg/opj_includes.h]: I saw that, I am building locally.

**Review by dzenanz — APPROVED** [review]

**Review by thewtex — APPROVED** [review]
@jamesobutler thanks!

**Comment by N-Dekker** [comment]
It seems to me that this commit _only_ defines `opj_lrintf` for MSVC and Borland:

https://github.com/InsightSoftwareConsortium/ITK/blob/7cebc26dfae73d90dbf4702aa1ccc544e9b70a37/Modules/ThirdParty/OpenJPEG/src/openjpeg/opj_includes.h#L91

The upstream header file also has an "else" for other compilers, at https://github.com/uclouvain/openjpeg/blob/c423cc84e7be79051a7f9631fa26aa7d072361f2/src/lib/openjp2/opj_includes.h#L157-L161
```
#else
static INLINE long opj_lrintf(float f) {
	return lrintf(f);
}
#endif
```

Not sure if this apparent discrepancy is of interest  🤔 because ITK does not call `opj_lrintf` anyway !

**Pull request merged**